File: day7/part1.py
# Read file input
with open("input.txt", "r") as f:
    data = [int(i) for i in f.read().rstrip('\n').split(',')]

data.sort()

target = data[len(data)//2] # Target position for crabs to align on (median position)

ans = 0
for pos in data:
    ans += abs(target-pos)

print(f"Result: {ans}")

File: day7/part1.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int main() {
	freopen("input.txt","r",stdin);

	// Process crab data
	vector<int> data;
	while (cin) {
		int fish;
		cin >> fish;
		data.push_back(fish);
		// Ignore next character (comma)
		getchar();
	}
	data.pop_back();

	sort(data.begin(),data.end());

	int target = data[data.size()/2]; // Target position for crabs to align on (median position)

	int ans = 0;
	for (int pos : data)
		ans += abs(target-pos);

	cout << "Result: " << ans << '\n';
}
